types/block_finality.go

```diff
@@ -1,60 +1,92 @@
 package types
 
 import (
+	"context"
 	"fmt"
 	"math/big"
+	"strconv"
 	"strings"
 
+	"github.com/agglayer/aggkit/log"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/rpc"
 	"github.com/invopop/jsonschema"
 )
 
+// BlockNumberFinality represents a block finality with an optional offset
 type BlockNumberFinality struct {
-	string `validate:"required"`
+	Block  BlockNumber
+	Offset int64
+	// IfNotFoundReturnsZeroFlag indicates whether to return zero if the block is not found
+	IfNotFoundReturnsZeroFlag bool
```

Comment:
I don't see this being used in any of the components. It looks like its always false.